src/protocols/config.js

```diff
@@ -30,3 +33,48 @@ export const hyperOptions = {
   // All options here: https://github.com/datproject/sdk/#const-hypercore-hyperdrive-resolvename-keypair-derivesecret-registerextension-close--await-sdkopts
   storage: DEFAULT_HYPER_DIR,
 };
+
+// Dynamically fetch RPC URL from default chain list
+let RPC_URL = null;
+
+(async () => {
```

Comment:
Why is this in an async function? Seems like it could cause race conditions. Might want to refactor so whatever needs to be initialized is done before setting up the protocol handlers